### DISCUSSION TITLE
Post your builds here!

### DISCUSSION THREAD

**Discussion by Dakkaron** [Dakkaron/Fairberry#23] [Show and tell]
Here's the right place to show off what you made!

**Comment by rustyhealey** [comment]
Here is mine - working, but not yet mounted in a case. 

For those trying this build, I had:
- 1x bad board from JLCPCB with a missing chip.
- 1x bad board from JLCPCB with a Hirose connector issue.

But the third board works great! I ordered enough of everything to make five, just in case. Glad I did. I haven't soldered in years. Took a good bit of practice to get comfortable again - though my soldering is still complete garbage.

I used this USB-C breakout:
https://www.adafruit.com/product/5978

I used makerfabs.com for the Q10 keyboards. Note these are used keyboards, which I was not expecting. One of the five I ordered was defective with failed (no clicky) keys. The other four appear fine, but I've only tested one.

I have an S23+ now which I am using for prototyping. My plan is to create a phone case incorporating both the phone and the keyboard components, made in two parts where the front and back halves screws together to allow the USB keyboard connector to stay permanently plugged in. 

If this works and is practical, then it might make sense to switch to a smaller format phone. 

I wanted to say THANK YOU to the team who figured all this out and made this possible. What a fun build so far.

![20241208_180644_s](https://github.com/user-attachments/assets/b68aab0d-6af4-4c86-b760-b490b8bc1f73)

**Reply by Dakkaron** [reply]
Really cool youbgot it working! Sorry to hear you got such bad luck with JLCPCB. For me it worked out fine every time, that's why I recommended them. But you aren't the first one talking about issues with them sadly...

If you keep the keyboard permanently plugged in, you'll need to charge wirelessly, but I guess that should be no issue.

Regarding the case, there's an OpenSCAD script in theis repo that you can use to generate a base case that you can adjust afterwards to fit your needs. JLCPCB also offers really cheap 3D printing services, though I haven't tried them yet.

Thanks a lot for the praise! Took me long enough to make all of this :)

**Reply by rustyhealey** [reply]
Quick update. Beginning to prototype case designs. Fitment is going to take a minute to get dialed in but I think I may be able to ultimately get the keyboard pretty flush with the screen. There's a screw-on "top" to this case a-la old school Otterbox, which is not shown in these photos.

On the S23+ it's going to be tall - wouldn't be my first choice of phone to daily-drive like this, but if this concept works out it ought to be great on a smaller format screen. My Key2 is 149mm high, while this S23+ is 159mm high. A standard S23 is only 146mm high, as a point of comparison. Plenty of phone options in/around that size class, which might actually be OK to hold with the keyboard attached permanently. 

![IMG_20241212_1246285](https://github.com/user-attachments/assets/f443424e-28ba-4a26-9bfa-89a3c0af8fab)
![IMG_20241212_1246203](https://github.com/user-attachments/assets/ee90b698-6f58-4396-abec-4dc4cc6068e8)

Photos taken with my Key2, hacked to run on Verizon, for the sake of nostalgia. 

I notice that with the keyboard plugged in, I get prompted periodically by Android for "which program" I want to have handle the device. There doesn't seem to be a way to permanently dismiss the prompt.

**Reply by Dakkaron** [reply]
Sorry for the late answer. The only way to really get rid of the prompt is to uninstall all apps that want to access device. Pretty lame that there's no better solution.

**Comment by XantiaActiva** [comment]
Soldering starts:
<img width="825" height="487" alt="grafik" src="https://github.com/user-attachments/assets/a4fe01b4-d0ed-41d0-843c-b10b299fcc71" />

Ready for programming:
<img width="822" height="531" alt="grafik" src="https://github.com/user-attachments/assets/17264613-a5ac-45b6-873a-144601b3e1fb" />

Burning the bootloader:
<img width="877" height="689" alt="grafik" src="https://github.com/user-attachments/assets/ca0c85e9-7d0c-4f06-91fe-c4eb1a10e468" />

USB port added:
<img width="1096" height="566" alt="grafik" src="https://github.com/user-attachments/assets/19690444-2f66-473d-bd85-2acb280a7fff" />

Uploading:
<img width="945" height="649" alt="grafik" src="https://github.com/user-attachments/assets/f179079e-d3e3-43f2-83f4-93571f989b9e" />

Done:
<img width="1099" height="176" alt="grafik" src="https://github.com/user-attachments/assets/7a2a26f8-d31d-4a2f-bd6c-586e995dd467" />

The case arrived. Its a little bit too wide, but the USB port will fix that.
<img width="563" height="878" alt="grafik" src="https://github.com/user-attachments/assets/f6bf5d00-3416-4d3a-a029-4e1ac68959a2" />

<img width="487" height="817" alt="grafik" src="https://github.com/user-attachments/assets/de180b15-a337-46e7-bebd-e4b96819829a" />

Keyboard fitted for test purposes:
<img width="567" height="838" alt="grafik" src="https://github.com/user-attachments/assets/9e2ef779-a82d-4674-ab1c-4bf93a53f437" />

Backside:
<img width="601" height="912" alt="grafik" src="https://github.com/user-attachments/assets/f80288c1-97d2-4f8d-836e-2a7dcf61ffa3" />

Not yet finally mounted:
<img width="963" height="495" alt="grafik" src="https://github.com/user-attachments/assets/82cbe7a3-d334-4aac-96c4-7127bcc5d423" />

Case with keyboard and circuit board:
<img width="459" height="466" alt="grafik" src="https://github.com/user-attachments/assets/9fea6632-0aec-4282-9f27-ea916036a0f7" />

At the moment the keyboard is only working if connected to a PC and when its connected with an USC-C OTG connector to the phone. Actually I am waiting for USB-C plugs (with OTG support) from China. The stuff sold here is Europe does not support OTG...

To be continued.

Best regards

**Reply by Dakkaron** [reply]
You can get OTG from european sellers too, but it really depends on the part.

This one here for example should work: https://www.amazon.de/dp/B07T97LC9L/

The important thing is that there's a 5.1k resistor for the CC lines to tell the phone that USB OTG should be enabled.

**Reply by XantiaActiva** [reply]
<img width="1127" height="399" alt="image" src="https://github.com/user-attachments/assets/f8421b43-2ef6-4a64-988a-1da92b965983" />

<img width="1213" height="910" alt="image" src="https://github.com/user-attachments/assets/897f5cb8-cf33-49bd-a367-50bd34b4e869" />

<img width="1213" height="910" alt="image" src="https://github.com/user-attachments/assets/61154921-7c41-4002-81eb-d694c5ffb5ab" />

<img width="1213" height="910" alt="image" src="https://github.com/user-attachments/assets/279f3d97-181c-41e6-baa2-97cacdd0a9ef" />

Finished so far. :)

**Reply by XantiaActiva** [reply]
Only issue is that the case is a little bit too wide. I could fit tighter.

**Reply by Dakkaron** [reply]
Nice! Yeah, dialing the fit in takes a few tries. It's also dependent on the tolerances of the 3D printer.

Btw, the bottom hole is not intentional, that looks like a bug. I don't have time to actually fix it right now, but as a workaround you can set

_ELECTRONICS_CUTAWAY_LENGTH=19;

instead of

_ELECTRONICS_CUTAWAY_LENGTH=25;

That should then print as expected.
